### PR TITLE
Pipelines and docs for CIS compliant tWAS VM Images as Azure Marketplace offerings

### PR DESCRIPTION
**Pull request by majguo**
## Description

The PR addresses most of the requirements listed in #64 by providing pipelines and docs for CIS compliant tWAS VM Images as Azure Marketplace offerings.

## Testing

CICD pipelines for generating VHD files for CIS compliant tWAS VM Images:
* [twas-base-cis CICD](https://github.com/majguo/azure.websphere-traditional.image/actions/runs/3957371392)
* [twas-nd-cis CICD](https://github.com/majguo/azure.websphere-traditional.image/actions/runs/3957422312)
* [ihs-cis CICD](https://github.com/majguo/azure.websphere-traditional.image/actions/runs/3956846632)

These generated VHD files are also used for updating the [testing tWAS VM offer](https://partner.microsoft.com/en-us/dashboard/commercial-marketplace/offers/387cd9d3-f4ca-454b-9a89-c612fffe29d6/overview) in PartnerCenter. After the [testing tWAS VM offer](https://partner.microsoft.com/en-us/dashboard/commercial-marketplace/offers/387cd9d3-f4ca-454b-9a89-c612fffe29d6/overview) is successfully published and in preview, it's successfully verified by deploying VM from both Azure Portal and solution template.

## How-To docs

Part of how-to docs of updating the VM images are located in the following PRs: 
* WASdev/azure.websphere-traditional.singleserver/pull/65
* WASdev/azure.websphere-traditional.cluster/pull/199

The others are included in this PR.

As a result, after the above PRs and this PR are merged, instructions on how to update CIS compliant tWAS VM images can be referenced from [howto-update-image.md](https://github.com/WASdev/azure.websphere-traditional.image/blob/cis/docs/howto-update-image.md).

### The remaining work

Depending on what images are used in the twas-single and twas-cluster solution template, e.g., 
* Only CIS compliant images are used
* Both CIS compliant and non-CIS compliant images, per user selection in UI

(**Note**: @edburns @m-reza-rahman @git4rk @gcharters) Pls lead the discussion to reach the consensus.)

the following sections need to be updated accordingly:
* [Updating and publishing the solution code for twas-single](https://github.com/WASdev/azure.websphere-traditional.singleserver/blob/main/docs/howto-update-for-was-fixpack.md#updating-and-publishing-the-solution-code)
* [Updating and publishing the solution code for twas-cluster](https://github.com/WASdev/azure.websphere-traditional.cluster/blob/main/docs/howto-update-for-was-fixpack.md#updating-and-publishing-the-solution-code)

Signed-off-by: Jianguo Ma <jiangma@microsoft.com>

### CODE REVIEW

File: .github/workflows/entitlement-check.yml [hunk 1]
```diff
@@ -80,13 +80,15 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: send notification
-        if: always()
+        if: ${{ env.msTeamsWebhook != 'NA' }}
```
Comment (edburns) [inline .github/workflows/entitlement-check.yml:83]: @majguo  love this. Thanks.

File: .github/workflows/ihs-cisBuild.yml [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+  # curl --verbose -X POST https://api.github.com/repos/${REPO_NAME}/dispatches -H "Accept: application/vnd.github.everest-preview+json" -H "Authorization: token ${PERSONAL_ACCESS_TOKEN}" --data '{"event_type": "integration-test-all"}'
+
+env:
+  repoName: azure.websphere-traditional.image
+  userName: ${{ secrets.USER_NAME }}
+  azureCredentials: ${{ secrets.AZURE_CREDENTIALS }}
+  entitledIbmUserId: ${{ secrets.ENTITLED_IBM_USER_ID }}
+  entitledIbmPassword: ${{ secrets.ENTITLED_IBM_USER_PWD }}
+  unEntitledIbmUserId: ${{ secrets.UNENTITLED_IBM_USER_ID }}
+  unEntitledIbmPassword: ${{ secrets.UNENTITLED_IBM_USER_PWD }}
```
Comment (edburns) [inline .github/workflows/ihs-cisBuild.yml:28]: @majguo I appreciate you already including the necessary setup and teardown scripts.

File: .github/workflows/ihs-cisBuild.yml [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+          curl -Lo external-deps-versions.properties https://raw.githubusercontent.com/Azure/azure-javaee-iaas/main/external-deps-versions.properties
+          source external-deps-versions.properties
+          echo "azCliVersion=${AZ_CLI_VERSION}" >> $GITHUB_ENV
+          echo "refArmttk=${ARM_TTK_REFERENCE}" >> $GITHUB_ENV
+          echo "refJavaee=${AZURE_JAVAEE_IAAS_REFERENCE}" >> $GITHUB_ENV
+      - name: Set up JDK 8
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'zulu'
+          java-version: '8'
```
Comment (edburns) [inline .github/workflows/ihs-cisBuild.yml:62]: I will add a commit to change this to microsoft and 8. Zulu is no longer recommended.

File: ihs-cis/src/main/scripts/was-check.sh [hunk 4]
```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+#      Copyright (c) Microsoft Corporation.
+# 
+#  Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (edburns) [inline ihs-cis/src/main/scripts/was-check.sh:5]: See https://github.com/WASdev/azure.websphere-traditional.image/issues/66.

**Review by edburns — APPROVED** [review]

**Review by git4rk — APPROVED** [review]

**Pull request merged**